ocaml/xapi/repository_helpers.ml

```diff
@@ -231,6 +231,19 @@ let assert_gpgkey_path_is_valid path =
     raise Api_errors.(Server_error (invalid_gpgkey_path, [path]))
   )
 
+let assert_remote_pool_url_is_valid ~url =
+  try
+    let uri = Uri.of_string url in
+    match (Uri.scheme uri, Uri.host uri, Uri.path uri) with
+    | Some "https", Some host, path
+      when path = Constants.get_enabled_repository_uri ->
+        Helpers.assert_is_valid_ip `ipv4or6 "url" host
+    | _ ->
+        raise Api_errors.(Server_error (internal_error, ["invalid url"]))
```

Comment:
This is not technically an internal error, it would be better to raise the invalid_base_url here, and reraise it in the with clause:
```ocaml
with e ->
  error "Invalid ...
  raise e
```